corehq/apps/users/bulkupload.py

```diff
@@ -438,8 +438,10 @@ def is_password(password):
                         error = custom_data_validator(data)
                         if error:
                             raise UserUploadError(error)
+                        data = {k: unicode(v) for k, v in data.iteritems()}
                         user.user_data.update(data)
                     if uncategorized_data:
+                        uncategorized_data = {k: unicode(v) for k, v in uncategorized_data.iteritems()}
```

Comment:
I'm assuming, after reading the FB case, this change is meant to coerce a non-string value (`int`, `float`, etc.) to a string. Notably, this will blow up if the value is a byte string that cannot be converted to Unicode with the default encoding (ASCII). I assume we're not worried about that here.